readme.md

```diff
@@ -246,11 +246,19 @@ When set to `true` the promise will return the [Response body](#body-1) instead
 
 ###### cookieJar
 
-Type: [`tough.CookieJar` instance](https://github.com/salesforce/tough-cookie#cookiejar)
+Type: `object` | [`tough.CookieJar` instance](https://github.com/salesforce/tough-cookie#cookiejar)
 
 **Note:** If you provide this option, `options.headers.cookie` will be overridden.
 
-Cookie support. You don't have to care about parsing or how to store them. [Example.](#cookies)
+Cookie support. You don't have to care about parsing or how to store them. [Example](#cookies).
+
+###### cookieJar.setCookie
+
+The function takes two arguments: `rawCookie` (`string`) and `url` (`string`). It needs to be an async function or some callback-style function.
```

Comment:
Let's only support async/promise-returning functions and just add a note that users needs to promisify callback-style functions themselves. Same with the below.

Reply:
Then we need a special case for `tough-cookie`, since they are callback-style.

Reply:
How do you want to detect `tough-cookie` since the dependency is optional? :P 
I mean we can support callback-style, but the readme would say the opposite.

Reply:
done